File: Visitor/Ruby/DebitCard.rb
require_relative "IPayment"
require_relative "CPU"
require_relative "VGA"

class DebitCard < IPayment
    def visit(component)
        if component.is_a?(CPU)
            puts "Purchase CPU with Debit Card"
        elsif component.is_a?(VGA)
            puts "Purchase VGA with Debit Card"
        end
    end
end

File: Visitor/Javascript/DebitCard.js
import IPayment from "./IPayment.js";
import CPU from "./CPU.js";
import VGA from "./VGA.js";

class DebitCard extends IPayment {
  visit(component) {
    if (component instanceof CPU) {
      console.log("Charging debit card for CPU");
    } else if (component instanceof VGA) {
      console.log("Charging debit card for VGA");
    }
  }
}

export default DebitCard;
